### ISSUE TITLE
Detected problems with API compatibility dialog shows starting Android P

### ISSUE THREAD

**Issue by amirh** [flutter/flutter#30395]
In https://github.com/flutter/engine/pull/8237 we used reflection to access hidden a11y APIs.
This landed as an interim fix that enables platform views accessibility up to Android O(starting Android P reflection access to system APIs is disallowed).
While the code is prepared for the reflection access to fail and should keep working(just without a11y support) no Android P and later, we got reports that Android is showing a warning dialog saying "https://github.com/flutter/engine/pull/8237" when running Flutter apps with that code.

I'm not sure what is the triggering mechanism for that warning dialog, and I couldn't get it to show on my Android P device.

I'm guessing that it is shown when a reflective access to a hidden API is executed, and so landed https://github.com/flutter/engine/pull/8393 to not even try these calls starting Android P.

https://github.com/flutter/engine/pull/8393 is a speculative fix as I could not confirm that it fixes the issue (as I still didn't get the dialog to show on my device).

Keeping this issue open until we confirm that the dialog is not showing anymore with the speculative fix.

**Comment by amirh** [comment]
Confirmed that if the reflection calls are not invoked the dialog shouldn't show, so flutter/engine#8393  should fix it. Closing this issue.

**Comment by BugsBunnyBR** [comment]
@amirh would you have any idea of how to remove the reclrctive calls and make the accessibility work in new android versions? Have you opened an issue in the android issue tracker or talked with android framework team about this? If there is not an issue or a consensus of how to proceed, how about to create the issue and document it somewhere?

**Comment by amirh** [comment]
This is tracked in: https://github.com/flutter/flutter/issues/19418
